lib/semian/net_http.rb

```diff
@@ -0,0 +1,90 @@
+require 'semian'
+require 'semian/adapter'
+
+module Net
+  ::Net::ProtocolError.include(::Semian::AdapterError)
```

Comment:
`https://github.com/ruby/ruby/blob/trunk/lib/net/http/responses.rb` shows the error codes and the exceptions they raise. `https://github.com/ruby/ruby/blob/trunk/lib/net/http/exceptions.rb` and `https://github.com/ruby/ruby/blob/trunk/lib/net/protocol.rb` lines 39-47 show all the exceptions being subclasses of `Net::ProtocolError`